templates/CRM/Financial/Form/BatchTransaction.tpl

```diff
@@ -188,6 +188,7 @@ function buildTransactionSelectorAssign(filterSearch) {
   "bDestroy"   : true,
   "bFilter"    : false,
   "bAutoWidth" : false,
+  "lengthMenu": [ 10, 25, 50, 100, 250, 500, 1000, 2000 ],
```

Comment:
I don't think so. I my experience the main risk of timeout is when the server is churning through millions of transactions for the selection. But even if the 2000 should cause troubles on weak servers/browsers users will quickly learn not to select it.

The main benefit is, that if you want to assign 10000 transactions to a group, you need a lot less steps (since there is no 'select all'),